modules/ROOT/pages/cors-policy.adoc

```diff
@@ -14,19 +14,118 @@ endif::[]
+
+The following diagram shows the XMLHttpRequest (XHR) in JavaScript flow for determining whether to execute the actual request:
+
+== CORS Algorithm
+
+The CORS policy is a two-step algorithm: 
+
+* If the request is considered to be ‘complex’ (see the previous client side diagram for XHR) then a preflight request needs to be executed. 
++
+If the server does not return the proper CORS response headers for the preflight, the client library (XHR in the previous example) is not allowed to perform the actual request.
+* If the preflight response is correct and complete, then the client library  executes the actual request, which includes certain CORS headers. 
++
+The client library then validates the CORS headers of the response. If some of the required headers are missing, then again the client library is obliged to block the response from reaching the client, which is usually a web page.
+
+The following section details the headers for the request and response for the CORS algorithm.
+
+== CORS Request Headers
+
+* Origin: The origin making the cross origin request.
+* Access-Control-Request-Method:  The method that is invoked in the actual request . This header is sent in the preflight request.
+* Access-Control-Request-Headers: Custom headers that are sent in the actual request. This header is sent in the preflight request.
+
+[NOTE]
+
+For the GET or HEAD methods, the standard defines a specific list of headers that are considered as ‘simple’ and that do not need to perform a preflight. For headers considered as ‘custom’  by the standard, a preflight is executed for GET and HEAD requests. See the example of XHR to validate the path in which the client does not have to perform a preflight.
+
+== CORS Response Headers
+	
+The headers included in a response differs based on whether the request is a preflight or an actual request:
+
+* Access-Control-Allow-Origin: Obligatory in every response.
++
+If this header is not present in the response, the browser or client library blocks the response from reaching the web page. The wildcard “*” can be used to represent any origin.
+* Access-Control-Allow-Methods: Allowed methods that can be executed.
++
+This headers is returned in the OPTIONS request (preflight). The server might respond with a list of allowed methods, delegating the validation task of whether its actual request is supported to the client.
+* Access-Control-Allow-Headers: Allowed headers in the actual request.
++ 
+This header works analogous to Access-Control-Allow-Methods.
+* Access-Control-Allow-Credentials: Notifies the client, with a Boolean value, whether the actual request can be made with a cookie.
```

Comment:
This is beginning to stretch the idea of a short text that can follow a bullet-list item and colon....

Reply:
Done! Thanks!